ianDavis/script.js

```diff
@@ -0,0 +1,335 @@
+
+//1)
+//a) Write a function fractionString(n,d) which takes 2 parameters (n,d) and returns a string. For example, fractionString(7,4) should return "1 3/4", and fractionString(3,3) should (for now) return "1 0/3". As before, assume that n and d are both positive integers.
+
+function fractionString(n, d) {
+	var wholeNumber = Math.floor(n /d);
+	if (n % d == 0) {
+		var fraction = "0" + "/" + d;
+	} else {
+		fraction = (n % d) + "/" + d;
+	}
+	return wholeNumber + " " + fraction;
+}
+
+//b) Write a second version of your function which improves the output in two special cases:
+
+function fractionString(n, d) {
+	var wholeNumber = Math.floor(n /d) || "";
+	if (n % d == 0) {
+		var fraction = "";
+	} else {
+		fraction = (n % d) + "/" + d;
+	}
+	return wholeNumber + " " + fraction;
+}
+
```

Comment:
Nit-picking,  but when you pass numbers such as (0, 4) the output is an empty string, when in this case it should probably output a '0'